compiler/rustc_trait_selection/src/traits/codegen.rs

```diff
@@ -120,13 +120,7 @@ where
     // In principle, we only need to do this so long as `result`
     // contains unbound type parameters. It could be a slight
     // optimization to stop iterating early.
-    let errors = fulfill_cx.select_all_or_error(infcx);
-    if !errors.is_empty() {
-        infcx.tcx.sess.delay_span_bug(
```

Comment:
This `delay_span_bug` can fail because of overzealous normalizations. I don't understand its role: is it correct to remove it?

Reply:
@matthewjasper, @nikomatsakis, do you have a suggestion?